examples/run_examples.sh

```diff
@@ -31,23 +31,28 @@ if ! python3 -c "import nest" >/dev/null 2>&1; then
     exit 1
 fi
 
-
 # set bash strict mode
 set -euo pipefail
+set -x
 IFS=$' \n\t'
 
+# current directory where all results are save is "basedir"
+basedir="$PWD"
+# source tree where all input scripts/data can be found is "sourcedir"
+sourcedir="$(realpath "$(dirname "$0")/..")"
+
 declare -a EXAMPLES
 if [ "${#}" -eq 0 ]; then
     # Find all examples that have a line containing "autorun=true"
     # The examples can be found in subdirectory nest and in the
     # examples installation path.
     EXAMPLELIST="$(mktemp examplelist.XXXXXXXXXX)"
     if [ -d "nest/" ] ; then
-        grep -rl --include='*.sli' 'autorun=true' nest/ >>"$EXAMPLELIST"
+        grep -rl --include='*.sli' 'autorun=true' "${sourcedir}/nest/" >>"$EXAMPLELIST" || true
```

Comment:
What is the purpose of `|| true`?

Reply:
If grep does not find the string (i.e. there are not exmples in that directory) then the script exits due to the error code. `true` always returns a successful return-code.